sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -26,40 +29,428 @@ class ConfigurationSetting(Model):
+                key_value.value = json.loads(key_value.value)
+            except json.decoder.JSONDecodeError:
+                pass
+        return cls(
+            key=key_value.key,
+            uri=key_value.value[u"uri"],
+            label=key_value.label,
+            secret_id=key_value.value,
             last_modified=key_value.last_modified,
             tags=key_value.tags,
             read_only=key_value.locked,
             etag=key_value.etag,
         )
+
+    def _to_generated(self):
+        # type: (...) -> KeyValue
+        return KeyValue(
+            key=self.key,
+            label=self.label,
+            value=json.dumps({u"uri": self.uri}),
+            content_type=self.content_type,
+            last_modified=self.last_modified,
+            tags=self.tags,
+            locked=self.read_only,
+            etag=self.etag,
+        )
+
+
+class FeatureFilterBase(object):
+    """Base class for the feature filters of FeatureFlagConfigurationSetting"""
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def _from_generated(cls, feature_filter):
+        if feature_filter["name"] == "Microsoft.Targeting":
+            return TargetingFeatureFilter.from_service(feature_filter["parameters"])
+        if feature_filter["name"] == "Microsoft.TimeWindow":
+            return TimeWindowFeatureFilter.from_service(feature_filter["parameters"])
+        if feature_filter["name"] == "Microsoft.Percentage":
+            return CustomFeatureFilter.from_service(feature_filter["parameters"])
+        raise ValueError(
+            "{} not recognized as a feature filter.".format(feature_filter["name"])
+        )
+
+    def _to_generated(self):
+        pass
+
+
+class TargetingFeatureFilter(FeatureFilterBase):
+    """A configuration setting that controls a feature flag
+    :param rollout_percentage:
+    :type rollout_percentage: int
+    :param users:
+    :type users: list[str]
+    :param groups:
+    :type groups: list[str]
+    """
+
+    def __init__(
+        self, rollout_percentage, users=None, groups=None
+    ):  # pylint:disable=super-init-not-called
+        # type: (str, dict) -> None
+        self._name = "Microsoft.Targeting"
+        self.rollout_percentage = rollout_percentage
+        self.users = users or []
+        self.groups = groups or []
+
+    @classmethod
+    def from_service(cls, dict_repr):
+        # type: (dict[str, str]) -> TargetingFeatureFilter
+
+        """Creates a TargetingFeatureFilter from the generated code call
+
+        :param dict_repr:
+        :type dict_repr: dict[str, str]
+        """
+
+        return cls(
+            dict_repr["Audience"]["DefaultRolloutPercentage"],
+            users=dict_repr["Audience"]["Users"],
+            groups=dict_repr["Audience"]["Groups"],
+        )
+
+    def _to_generated(self):
+        # type: (...) -> dict
+        return {
+            "name": self._name,
+            "parameters": {
+                "Audience": {
```

Comment:
The schema seems to be different from https://apiview.dev/Assemblies/Review/d89b0cb38b474b07ad545fd89df0a4a2?diffRevisionId=ead773d194c94a3f8e72a482135355e6&doc=False&diffOnly=False#@azure/app-configuration!FeatureFlagTargetingClientFilter:interface


Sample JSON
```
{
   "id":"MyFeature",
   "description":"",
   "enabled":true,
   "conditions":{
      "client_filters":[
         {
            "name":"Microsoft.Targeting",
            "parameters":{
               "Audience":{
                  "Users":[
                     "sadf",
                     "aaa"
                  ],
                  "Groups":[
                     {
                        "Name":"asdf",
                        "RolloutPercentage":100
                     },
                     {
                        "Name":"fdas",
                        "RolloutPercentage":29
                     }
                  ],
                  "DefaultRolloutPercentage":72
               }
            }
         }
      ]
   }
}
```